plugins/extensions/plugins/DeflectPlugin.cpp

```diff
@@ -20,188 +20,197 @@
+                _phi += frameBuffer.getSize().y() / 100.f * std::asin( dv );
+
+                _previousTouchPosition.x() = event.mouseX;
+                _previousTouchPosition.y() = event.mouseY;
+                _previousTouchPosition.z() = std::min(0.f, _previousTouchPosition.z());
             }
+            else
+                _previousTouchPosition.z() += event.dy;
+
+            Scene& scene = _brayns.getScene();
+            const Vector3f& center = scene.getWorldBounds().getCenter();
+            const Vector3f& size = scene.getWorldBounds().getSize();
+            const Vector3f cameraPosition = center + size * Vector3f(
+                _previousTouchPosition.z( ) * std::cos( _phi ) * std::cos( _theta ),
+                _previousTouchPosition.z( ) * std::sin( _phi ) * std::cos( _theta ),
+                _previousTouchPosition.z( ) * std::sin( _theta ));
+
+            _brayns.getCamera().setPosition( cameraPosition );
+            _brayns.getCamera().setTarget( center );
+            break;
+        }
+        case deflect::Event::EVT_KEY_PRESS:
+        {
+            _brayns.getEngine().getKeyboardHandler()->handleKeyboardShortcut( event.text[0] );
+            break;
+        }
+        case deflect::Event::EVT_VIEW_SIZE_CHANGED:
+        {
+            _brayns.reshape( Vector2ui( event.dx, event.dy ) );
+            break;
         }
+        case deflect::Event::EVT_NONE:
+        case deflect::Event::EVT_CLICK:
+        default:
+            break;
+        }
+    }
+    if( hasEvents )
+    {
+        _brayns.getFrameBuffer().clear();
+        _brayns.getRenderer().commit();
     }
 }
 
 void DeflectPlugin::_send(
     const Vector2i& windowSize,
     unsigned long* imageData,
-    const bool swapXAxis)
+    const bool swapXAxis )
 {
-    if(!_stream->isConnected())
-        return;
+    deflect::ImageWrapper deflectImage( imageData, windowSize.x(), windowSize.y(), deflect::RGBA );
 
-    deflect::ImageWrapper deflectImage(
-                imageData, windowSize.x(), windowSize.y(), deflect::RGBA);
+    ApplicationParameters& applicationParameters =
+        _brayns.getParametersManager().getApplicationParameters();
 
+    deflectImage.compressionQuality = applicationParameters.getJpegCompression();
     deflectImage.compressionPolicy =
-            _compressImage ? deflect::COMPRESSION_ON : deflect::COMPRESSION_OFF;
+        ( deflectImage.compressionQuality != 100 ) ?
+        deflect::COMPRESSION_ON : deflect::COMPRESSION_OFF;
 
-    deflectImage.compressionQuality = _compressionQuality;
     if( swapXAxis )
-        deflect::ImageWrapper::swapYAxis(
-            (void*)imageData, windowSize.x(), windowSize.y(), 4);
+        deflect::ImageWrapper::swapYAxis( (void*)imageData, windowSize.x(), windowSize.y(), 4);
 
-    bool success = _stream->send(deflectImage);
+    // TODO: Big Memory Leak on every send. FIX NEEDED!!!!!
+    const bool success = _stream->send( deflectImage );
     _stream->finishFrame();
```

Comment:
also evaluate result of finishFrame(). We typically do result = send() && finish();